### ISSUE TITLE
UnifiedMap: Poss. "refresh problem" on changing target

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#15473]
_Originally posted by @MagpieFourtyTwo in https://github.com/cgeo/cgeo/issues/15470#issuecomment-2002105630_

Just remembered another poss. "refresh problem" with Unified Map today:
- had an active route to a cache
- then changed the routing target to it's parking lot (by clicking the [P] icon and then the routing icon in the popup header)
- but ... route did not change at all (although I tried several times)

Maybe over there a refresh is missing, too?

**Comment by Lineflyer** [comment]
I observed such issue as well, but for me it was a question of either time or movement. Meaning: Once I changed my position the route changed over to the new target. Or maybe it was not due to my movement but it just needed some seconds.

**Comment by moving-bits** [comment]
@MagpieFourtyTwo
Which type of "route" are you talking of - individual route ("blue line"), maybe with activated "set route as target" (or not), or direction line ("red line"), or a mix of both?

**Comment by moving-bits** [comment]
When using direction lines, it seems to work over here:
- select a cache and tap the navigation icon in the info sheet => cache coords get set as target, red/pink line appears
- select the cache's parking waypoint and tap the navigation icon in the info sheet => coords of parking waypoint get set as target, and red/pink line switches to the waypoint

**Comment by MagpieFourtyTwo** [comment]
> Or maybe it was not due to my movement but it just needed some seconds.

As already mentioned in other issues, I too have already observed several times that routing may take *a lot* longer in UM.
Having this in mind, I gave c:geo up to 20 seconds (and in fact the same amount of steps) to recalculate (resp. display) the route to the new target - but nothing happened (apart from the fact that the history track flashed once per step).

Even stranger, as the route in question was not long at all:  I just switched from my current walking route (about 500 meters of walking distance to the first target) to the parking lot, which had about 200 meters more.

**Comment by MagpieFourtyTwo** [comment]
> Which type of "route" are you talking of - individual route ("blue line"), maybe with activated "set route as target" (or not), or direction line ("red line"), or a mix of both?

"Normal" direction line, like described in the OP: 
1. clicked on a cache icon and then on the routing symbol in the bottom sheet header (first, initial route)
2. clicked on the parking lot icon of this cache and then on the routing symbol in it's bottom sheet header.

In fact the exact same way you stated as working in [issuecomment-2002108375](https://github.com/cgeo/cgeo/issues/15473#issuecomment-2002108375).  Strange.

**Comment by moving-bits** [comment]
> In fact the exact same way you stated as working in [issuecomment-2002108375](https://github.com/cgeo/cgeo/issues/15473#issuecomment-2002108375). Strange.

I had checked with `release`, maybe there's a difference to `master` - will cross-check tomorrow.

**Comment by MagpieFourtyTwo** [comment]
Things got even worse today, even *so* bad that I finally had to deactivate UM to get navigation going.
Scenario:
1. opened listing of an event (that took place somewhat off the beaten track)
2. from the listing page clicked ``Navigate to``
3. navigation draw a straight line to the event coords (as was to be expected)
4. clicked on the parking lot, which was an official [P] and close to a big road
5. route flickered once, but target stayed to be the event listing coords
6. clicked a second time on the parking lot
7. route finally switched to the parking lot, but as soon as I started to move, it switched back to the event listings

This game happened several times. Even after fresh restart of c:geo.  Most of the times in point 7. it did not even need a movement, just a bit of time (about 5 to 10 secs) and the route jumped back to a straight line to the event listings by itself. And even if it stayed, it switched back the latest after going to Android's home screen and back to c:geo. Changing routing style did either switch back, too, or did just nothing ... same with moving the map ...

Switched back to classic map and the problem was gone. And as I was a little late already, I stayed with classic. Once more ...  :/

Explanation for the attached (compressed) video:
0:00 route to event ... results in a straight (as expected)
0:02 changed the route to parking lot - nothing happens, apart from route flickering once
0:08 second try to change route target to [P] - route switches immediately (as should have done on first try)
waited till 0:20 if route will switch again, but did not do the favor ;)
0:19 switched routing style to bike - individual changes, but navigation switches back to event coords (target is still [P]!)
0:26: set rout target to [P] again (although it still should be the target)
0:29 now route switches to "Bike" with target [P] (as should have done at 0:19)
0:33 changing the routing style to "Car" - but route remains the same
0:41 just moving the map a bit - route switches back as a straight line to event coords
0:45 again set route target to [P] (which it should have been since 0:26 already) and now route is shown correctly
0:53 switching routing style to "Bike" - takes 10 seconds to show the new route (although its length is just 17 km!)
1:06 switching target to [P] again ... route immediately changes
1:12 switching routing style to "Car"- nothing happens ...
1:21 as soon as I started to move target switched back to event coords, while current target should still be [P]

All this, BTW after a fresh restart of c:geo.


https://github.com/cgeo/cgeo/assets/9889015/4516d6e6-3d8b-44c3-a885-1038e4c0301a
